lms/services/lti_registration.py

```diff
@@ -27,6 +27,40 @@ def get(self, issuer: str, client_id: Optional[str] = None):
 
         return query.one_or_none()
 
+    def get_by_id(self, id_) -> LTIRegistration:
+        return self._db.query(LTIRegistration).get(id_)
+
+    def search_registrations(self, *, issuer=None, client_id=None) -> LTIRegistration:
+        """Return the registrations that match all of the passed parameters."""
+
+        query = self._db.query(LTIRegistration)
+        if issuer:
+            query = query.filter_by(issuer=issuer)
+
+        if client_id:
+            query = query.filter_by(client_id=client_id)
+
+        return query.all()
+
+    def create_registration(  # pylint:disable=too-many-arguments
+        self,
+        issuer: str,
+        client_id: str,
+        auth_login_url: str,
+        key_set_url: str,
+        token_url: str,
```

Comment:
I'm not sure basic types add a lot. They don't cause much trouble either, but I think it might harm readability vs.

```python
def create_registration(  # pylint:disable=too-many-arguments
        self, issuer, client_id, auth_login_url, key_set_url, token_url
    ):
```